### PR TITLE
Fix `mul_by_inverse` for constants

### PR DESCRIPTION
**Pull request by Pratyush**

### CODE REVIEW

File: src/fields/mod.rs [hunk 1]
```diff
@@ -160,14 +160,18 @@ pub trait FieldVar<F: Field, ConstraintF: Field>:
...
     fn mul_by_inverse(&self, denominator: &Self) -> Result<Self, SynthesisError> {
-        let result = Self::new_witness(self.cs(), || {
-            let denominator_inv_native = denominator.value()?.inverse().get()?;
-            let result = self.value()? * &denominator_inv_native;
-            Ok(result)
-        })?;
-        result.mul_equals(&denominator, &self)?;
+        if self.is_constant() && denominator.is_constant() {
+            Ok(self.clone() * denominator.value()?.inverse().unwrap())
+        } else {
```
Comment (ValarDragon) [inline src/fields/mod.rs:165]: This doesn't depend on self being constant right?

Comment (Pratyush) [reply src/fields/mod.rs:165]: So you're saying I change it to
```rust
	if denominator.is_constant() {
		Ok(denominator.inverse()? * self)
	} else {
```

Comment (Pratyush) [reply src/fields/mod.rs:165]: Done

Comment (ValarDragon) [reply src/fields/mod.rs:165]: yeah

**Comment by Pratyush** [comment]
Is there anything else to be done here, @ValarDragon ?

**Pull request merged**